File: CS/DataLayer/Data/RandomDataGenerator.cs
using DataLayer.Data.Classes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DataLayer.Data
{
    public static class RandomDataGenerator
    {
        public static void GenerateRandomData(XtraGridTutorialDBContext context)
        {
            Random rand = new Random();
            context.Categories.Add(new Category { CategoryName = "Entertainment", TransactionType = TransactionType.Expense });
            context.Categories.Add(new Category { CategoryName = "Groceries", TransactionType = TransactionType.Expense });
            context.Categories.Add(new Category { CategoryName = "Automobile Expenses", TransactionType = TransactionType.Expense });
            context.Categories.Add(new Category { CategoryName = "Gifts", TransactionType = TransactionType.Expense });
            context.Categories.Add(new Category { CategoryName = "Loans", TransactionType = TransactionType.Expense });
            context.Categories.Add(new Category { CategoryName = "Phone", TransactionType = TransactionType.Expense });
            context.Categories.Add(new Category { CategoryName = "Taxes", TransactionType = TransactionType.Expense });
            context.Categories.Add(new Category { CategoryName = "Utilities", TransactionType = TransactionType.Expense });
            context.Categories.Add(new Category { CategoryName = "Money Transfer", TransactionType = TransactionType.Expense });
            context.Categories.Add(new Category { CategoryName = "Salary", TransactionType = TransactionType.Income });
            context.Categories.Add(new Category { CategoryName = "Small Business", TransactionType = TransactionType.Income });
            context.Categories.Add(new Category { CategoryName = "Other", TransactionType = TransactionType.Income });
            context.Categories.Add(new Category { CategoryName = "Money Transfer", TransactionType = TransactionType.Income });
            context.Accounts.Add(new Account { AccountName = "Cash" , Amount = rand.Next(1000) });
            context.Accounts.Add(new Account { AccountName = "Salary Card", Amount = rand.Next(5000) });
            context.Accounts.Add(new Account { AccountName = "Credit Card", Amount = rand.Next(5000) });
            context.SaveChanges();
           
            foreach (Account account in context.Accounts)
                foreach (Category category in context.Categories)
                    for (int i = 0; i < rand.Next(10); i++)
                    {
                        var transactionType = rand.Next(10) > 5 ? TransactionType.Income : TransactionType.Expense;
                        context.Transactions.Add(new Transaction
                        {
                            Account = account,
                            Category = category,
                            TransactionType = transactionType,
                            Amount = (transactionType == TransactionType.Income) ? rand.Next(10, 1000) : rand.Next(-1000, -10),
                            Comment = "Test transaction",
                            TrasactionDate = DateTime.Today.AddDays(-rand.Next(1000))
                        });    
                    }
                    
            context.SaveChanges();
        }

    }
}

File: VB/DataLayer/Data/RandomDataGenerator.vb
Imports DataLayer.Data.Classes
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace DataLayer.Data
    Public NotInheritable Class RandomDataGenerator

        Private Sub New()
        End Sub

        Public Shared Sub GenerateRandomData(ByVal context As XtraGridTutorialDBContext)
            Dim rand As New Random()
            context.Categories.Add(New Category With { _
                .CategoryName = "Entertainment", _
                .TransactionType = TransactionType.Expense _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Groceries", _
                .TransactionType = TransactionType.Expense _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Automobile Expenses", _
                .TransactionType = TransactionType.Expense _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Gifts", _
                .TransactionType = TransactionType.Expense _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Loans", _
                .TransactionType = TransactionType.Expense _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Phone", _
                .TransactionType = TransactionType.Expense _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Taxes", _
                .TransactionType = TransactionType.Expense _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Utilities", _
                .TransactionType = TransactionType.Expense _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Money Transfer", _
                .TransactionType = TransactionType.Expense _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Salary", _
                .TransactionType = TransactionType.Income _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Small Business", _
                .TransactionType = TransactionType.Income _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Other", _
                .TransactionType = TransactionType.Income _
            })
            context.Categories.Add(New Category With { _
                .CategoryName = "Money Transfer", _
                .TransactionType = TransactionType.Income _
            })
            context.Accounts.Add(New Account With { _
                .AccountName = "Cash", _
                .Amount = rand.Next(1000) _
            })
            context.Accounts.Add(New Account With { _
                .AccountName = "Salary Card", _
                .Amount = rand.Next(5000) _
            })
            context.Accounts.Add(New Account With { _
                .AccountName = "Credit Card", _
                .Amount = rand.Next(5000) _
            })
            context.SaveChanges()

            For Each account As Account In context.Accounts
                For Each category As Category In context.Categories
                    Dim i As Integer = 0
                    Do While i < rand.Next(10)
                        Dim transactionType = If(rand.Next(10) > 5, DataLayer.Data.Classes.TransactionType.Income, DataLayer.Data.Classes.TransactionType.Expense)
                        context.Transactions.Add(New Transaction With { _
                            .Account = account, _
                            .Category = category, _
                            .TransactionType = transactionType, _
                            .Amount = If(transactionType = DataLayer.Data.Classes.TransactionType.Income, rand.Next(10, 1000), rand.Next(-1000, -10)), _
                            .Comment = "Test transaction", _
                            .TrasactionDate = Date.Today.AddDays(-rand.Next(1000)) _
                        })
                        i += 1
                    Loop
                Next category
            Next account

            context.SaveChanges()
        End Sub

    End Class
End Namespace
